docs/product/specs/permissions-revamp.md

```diff
@@ -0,0 +1,115 @@
+  - This applies for explorations and metadata such as table and column settings.
+  - Future consideration:
+    - We can allow an additional option 'Infer from DB privileges' that will automatically identify access to each Mathesar-object based on the privileges on the underlying DB objects.
+
+### Schema level access control options
+* If a user has ownership access (or appropriate GRANT priviliges) to a schema based on their underlying role, they should be able to grant/revoke access to that schema to other roles configured in Mathesar.
+* While granting access to a role, the user can choose access control at three different levels.
+  - manage: USAGE, GRANT, SELECT, INSERT, UPDATE, DELETE, SET and ALTER
+  - edit: USAGE, SELECT, INSERT, UPDATE, DELETE, SET and ALTER
+  - view: USAGE, SELECT
+* This setting will iteratively apply to all objects within the schema.
+* For schemas created using the Mathesar UI, we will automatically grant view access to all configured roles, manage access to the 'default role', and we will provide access accordingly for auto-generated roles.
+* This will not be present for the 'public schema'.
+  - It should be made clear to the user that all users can access the public schema.
+* The UX should make it clear that the roles are the ones being granted access, and should display a list of users per role to make it easier for the user to understand which Mathesar users get access.
+* In the future, this option will be extended to all DB objects such as tables and views.
+
+### Default role
+* There should be one default role per database.
+* When a database is first configured, the role used to connect to the database will be set as the default role.
+* When new admins are added to an existing Mathesar instance which contains a number of databases, they will have access to all those databases by utilizing default roles (which are configured per database).
+* It is also used to deal with ownership issues, mentioned later in this document.
+* The default role cannot be changed. 
+  - We can provide this option in the future.
+  - This restriction is to reduce scope since changing the default role has more considerations such as changing ownerships.
+* The other way to think of this role is as 'Mathesar server's role to connect to the database for operations that do not directly involve a user'.
+
+### Dealing with Ownership and Collaboration issues
+* PostgreSQL has ownership for all DB objects.
+* Tables created by one role cannot be viewed/accessed by another role, unless the access is granted explicitly.
+* When tables/schemas are created via Mathesar UI, it will cause collaboration issues among Mathesar users if they utilize different roles.
+* We will be handling this by following the steps below:
+  * When a DB object is created via the UI, the ownership is transferred to the 'default role'.
```

Comment:
What about simply granting any role created or managed via Mathesar to the 'default role'? If you transfer ownership to the 'default role', you won't be able to `ALTER` or `DROP` a table you yourself created, regardless of your other privileges.

Reply:
I'm thinking the other way around in this case. If we configure the role `brent` to be used from Mathesar, then we'd `GRANT brent TO default_role`, not `GRANT default_role TO brent`.

With that said, I had a chat with Pavish today, and my (our) thinking has evolved a bit.

Reply:
> I'm thinking the other way around in this case

Ah right. Sorry — I misread your comment.

Reply:
This is no longer relevant.

We've introduced displaying the ownership for objects and there are changes revolving granting of privileges in https://github.com/mathesar-foundation/mathesar-wiki/commit/2ad0d7262212be9b5897deda8b7962e99d2f7191.